fili-generic-example/src/main/java/com/yahoo/wiki/webservice/application/GenericBinderFactory.java

```diff
@@ -0,0 +1,87 @@
+import com.yahoo.bard.webservice.druid.client.DruidWebService;
+import com.yahoo.bard.webservice.table.PhysicalTableDictionary;
+import com.yahoo.wiki.webservice.data.config.auto.DataSourceConfiguration;
+import com.yahoo.wiki.webservice.data.config.auto.DruidNavigator;
+import com.yahoo.wiki.webservice.data.config.dimension.GenericDimensions;
+import com.yahoo.wiki.webservice.data.config.metric.GenericMetricLoader;
+import com.yahoo.wiki.webservice.data.config.table.GenericTableLoader;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Builds dimensions, metrics, and tables for all datasources found from druid.
+ */
+public class GenericBinderFactory extends AbstractBinderFactory {
+    private final Supplier<List<? extends DataSourceConfiguration>> configLoader;
+    private GenericDimensions genericDimensions;
+    private static Set<DimensionConfig> dimensions;
+
+    /**
+     * Constructs a GenericBinderFactory which starts configuring with druid.
+     */
+    public GenericBinderFactory() {
+        DruidWebService druidWebService = buildMetadataDruidWebService(getMappers().getMapper());
+        configLoader = new DruidNavigator(druidWebService);
+        configLoader.get();
+    }
+
+    /**
+     * Provides a way for {@link GenericMain} to access the dimensions.
+     * TODO: This is a bad solution to getting the dimensions from GenericMain
+     * @return the dimensions found from druid.
+     */
+    public static Set<DimensionConfig> getDimensions() {
+        return dimensions;
+    }
+
+    @Override
+    protected Set<DimensionConfig> getDimensionConfigurations() {
+        //NOTE: This is guaranteed to be called before getTableLoader()
+        genericDimensions = new GenericDimensions(configLoader);
+        dimensions = genericDimensions.getAllDimensionConfigurations();
+        return genericDimensions.getAllDimensionConfigurations();
+    }
+
+    @Override
+    protected TableLoader getTableLoader() {
+        return new GenericTableLoader(configLoader, genericDimensions, getDataSourceMetadataService());
+    }
+
+    @Override
+    protected DruidDimensionsLoader buildDruidDimensionsLoader(
+            DruidWebService webService,
+            PhysicalTableDictionary physicalTableDictionary,
+            DimensionDictionary dimensionDictionary
+    ) {
+        List<List<Dimension>> dimensionsList = genericDimensions.getAllDimensionConfigurations().stream()
+                .map(DimensionConfig::getApiName)
+                .map(dimensionDictionary::findByApiName)
+                .map(Collections::singletonList)
+                .collect(Collectors.toList());
+        return new DruidDimensionsLoader(
+                webService,
+                dimensionsList,
+                DruidDimensionsLoader.buildDataSourcesList(physicalTableDictionary)
```

Comment:
Making this change would be a difference in how the `DruidDimensionLoader` has worked, but it _should_ be OK to do (I think), but we'll want to check what the defaults have been and what it has been doing from a behavior standpoint. 

I think the best approach (and we've started to move this way) is to have 2 constructors:

1. That takes the list of dimensions (ie. no dependencies on SystemConfig at all)
2. One that loads the dimensions from the configuration, resolves them, and then calls to the other constructor that takes the list.

The advantages of this are multiple, but here are a few:

1. Easier to test

    If we have to go through SystemConfig, it's harder to test because we have to set up the dimensions via both being in the dictionary, as well as set and revert a config parameter. If we can just pass in the Dimensions to load, it's much simpler

2. More reusable

    If we have a different source for what dimensions to load (ie. the ones that we're pulling in from Druid), then that use-case can just call the "primary" constructor directly and pass in the Dimensions desired.

If I'm remembering correctly (it's been a long weekend), this PR pushes us in the direction I was just talking about, though perhaps it doesn't go quite all the way there?